packages/backend-api/src/test/integration/websocket/assign_moderators.spec.ts

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as chai from 'chai';
+import * as WebSocket from 'ws';
+
+import { clearInterested, makeServer } from '@conversationai/moderator-backend-core';
+
+import {destroyUpdateNotificationService} from '../../../api/services/updateNotifications';
+import {mountAPI} from '../../../index';
+import {
+  expect,
+  makeArticle,
+  makeCategory,
+  makeUser,
+  sleep,
+} from '../../test_helper';
+
+const chaiHttp = require('chai-http');
+chai.use(chaiHttp);
+
+describe('websocket tests: assign moderators', () => {
+  let app: any;
+  let server: any;
+  let socket: any;
+  let gotClose = false;
+  let gotMessage = 0;
+
+  let user: any;
+  let category: any;
+  let article: any;
+
+  beforeEach(async () => {
+    user = await makeUser();
+
+    category = await makeCategory();
+    article = await makeArticle({categoryId: category.id});
+
+    const serverStuff = makeServer(true);
+    app = serverStuff.app;
+    app.use('/', (req: any, _: any, next: () => void) => {
+      req.user = user;
+      next();
+    });
+
+    app.use('/', mountAPI(true));
+    server = serverStuff.start(3000);
+    socket = new WebSocket('ws://localhost:3000/services/updates/summary');
+    socket.onclose = () => {
+      gotClose = true;
+    };
+
+    socket.onmessage = () => {
+      gotMessage += 1;
+    };
+    await sleep(500);
+    expect(gotClose).is.false;
+    expect(gotMessage).is.equal(2);
+    gotMessage = 0;
+  });
+
+  afterEach(async () => {
+    await socket.close();
+    await server.close();
+    await clearInterested();
+    await sleep(500);
```

Comment:
I'm a bit worried about long sleeps in test; do you really need this? We should be thinking about how to have promises that resolve when stuff is done rather than introduce sleeps like this.

Reply:
Maybe add a TODO? And think about how to improve going forward.